lib/periodic_jobs.rb

```diff
@@ -235,13 +237,14 @@
   # Send Decision Review emails to Veteran for failed form/evidence submissions
   mgr.register('5 1 * * *', 'DecisionReview::FailureNotificationEmailJob')
 
-  # Daily 0000 hrs job for Vye: performs ingress of state from BDN & TIMS.
-  mgr.register('15 00 * * 1-5', 'Vye::MidnightRun::IngressBdn')
-  mgr.register('45 03 * * 1-5', 'Vye::MidnightRun::IngressTims')
-
-  # Daily 0600 hrs job for Vye: activates ingressed state, and egresses the changes for the day.
-  mgr.register('45 05 * * 1-5', 'Vye::DawnDash')
-
-  # Daily job for Vye: clears deactivated BDNs every evening.
-  mgr.register('00 19 * * 1-5', 'Vye::SundownSweep')
+  # Skip running VYE jobs on federal holidays
+  unless Holidays.on(Time.zone.today, :us, :observed).any?
```

Comment:
Due to this running at process boot, I don't expect this will work as expected. I also don't know how schedules get removed as there only seems to be an interface for registering.

It would probably be better to move this to check inside the job.

Here's some generated solutions:

1. **Move Holiday Check Inside Job Classes**:
```ruby
class Vye::MidnightRun::IngressBdn
  include Sidekiq::Worker

  def perform
    return if holiday?
    # actual job logic
  end

  private

  def holiday?
    Holidays.on(Time.zone.today, :us, :observed).any?
  end
end
```

2. **Use Sidekiq Enterprise's Scheduler Window** (if you have Enterprise):
```ruby
class Vye::MidnightRun::IngressBdn
  include Sidekiq::Worker
  
  sidekiq_options scheduler_window: -> { 
    Holidays.on(Time.zone.today, :us, :observed).any? ? 0 : 24 
  }
end
```

3. **Use a Job Middleware**:
```ruby
class HolidayMiddleware
  def call(worker, msg, queue)
    if Holidays.on(Time.zone.today, :us, :observed).any?
      logger.info "Skipping #{worker.class} due to holiday"
      return
    end
    
    yield
  end
end

# In config/initializers/sidekiq.rb
Sidekiq.configure_server do |config|
  config.server_middleware do |chain|
    chain.add HolidayMiddleware
  end
end
```

4. **Use a Custom Schedule Module**:
```ruby
module BusinessDayScheduler
  def self.register_business_day_job(mgr, cron, worker_class)
    mgr.register(cron, worker_class) do |msg|
      # Only schedule if it's not a holiday
      !Holidays.on(Time.zone.today, :us, :observed).any?
    end
  end
end

# Usage:
BusinessDayScheduler.register_business_day_job(
  mgr,
  '15 00 * * 1-5',
  'Vye::MidnightRun::IngressBdn'
)
```

The middleware approach (#3) is probably the most flexible and maintainable because:

- It centralizes holiday logic in one place
- It's runtime-aware (checks holidays when jobs actually run)
- It can be easily enabled/disabled per worker if needed
- It provides consistent logging and monitoring
- It doesn't require modifying existing job classes

You could enhance the middleware further:

```ruby
class HolidayMiddleware
  def call(worker, msg, queue)
    return yield if worker.class.skip_holiday_check
    
    if holiday?
      Sidekiq.logger.info "Skipping #{worker.class} due to holiday: #{current_holiday}"
      return
    end
    
    yield
  end

  private

  def holiday?
    current_holiday.present?
  end

  def current_holiday
    @current_holiday ||= Holidays.on(Time.zone.today, :us, :observed).first
  end
end

# Optional: Add ability to skip holiday check for specific workers
module HolidayAware
  def self.included(base)
    base.extend ClassMethods
  end

  module ClassMethods
    def skip_holiday_check
      @skip_holiday_check ||= false
    end

    def skip_holiday_check=(value)
      @skip_holiday_check = value
    end
  end
end
```

Then in your job classes:

```ruby
class Vye::MidnightRun::IngressBdn
  include Sidekiq::Worker
  include HolidayAware
  
  # Optionally override for specific jobs
  # self.skip_holiday_check = true
end
```

This gives you a robust, centralized way to handle holiday job scheduling with good logging and the flexibility to override the behavior when needed.